.github/workflows/build-docker.yml

```diff
@@ -0,0 +1,23 @@
+name: Docker Image CI
+
+on:
+  push:
```

Comment:
As I understand it, the output of this workflow should be a versioned container image uploaded to Docker Hub, and it's good practice that each version should correspond with exactly one tag in the `git` repository (so that people can find the code used by the container - and potentially view relevant changelogs/diffs before upgrading).

That makes me wonder if we might want to consider filtering the `on` condition to the [`create[ref_type=tag]`](https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#create) event.  What do you reckon?

Reply:
sounds good, but then we need to change the tag from docker-compose to be an argument, so I would recommend that: https://docs.docker.com/compose/environment-variables/#substitute-environment-variables-in-compose-files. I will do the change